[Ong Seeu Sim] iP

SeeuSim opened the pull request:
# Rick and ~~Morty~~ Duke

>Listen to me, Morty. I know that new situations can be intimidating. You lookin’ around and it’s all scary and different, but y’know … 
> meeting them head-on, charging into ‘em like a bull — that’s how we grow as people.  - Rick Sanchez ([source](https://www.adultswim.com/videos/rick-and-morty))

Having issues with your memory? Always been wanting to do things, but somehow they are *erased* from your mind?
Well, that's because you're not using Rick and ~~Morty~~ ***Duuuke***

It's spectacular!

- All you need are is *text* commands
- So easy, Morty knows how to use it
- Blazing fast (it's almost like a portal transported you ahead in time)

If Morty is smarter than you, here's the guide he wrote:

1. Download it from [here](#)
2. Click on the app icon (Rick's butt)
3. Tell Rick what you need to remember
4. He may or *may not* remind you of those 🚩🙃

Here's a list of what I provide:
- [x] Adding, deleting tasks
- [x] Marking tasks as ***done***/***undone***
- [x] Description text search ***AND*** deadline based filtering
- [ ] Notifications (Work in progress)
- [ ] Modifying deadlines (Morty doesn't know how to do it)  

Here's how to interact with it (*hint*: you ***type*** in the app):
```shell
todo {task description}
```

If you wish to add more commands, simply add a Command class in the codebase:
```java
class MortyCommand extends Command {
    @Override
    public String execute(Tasklist ts, Ui ui) {
        ...
    }
}
```
And modify the [`Parser.java`](https://github.com/SeeuSim/ip/blob/master/src/main/java/duke/Parser.java) class to reference and create your command:
```java
//Parser::simpleCommand or Parser::twoArgCommand:
public static Command simpleCommand(String cmd) {
    switch(cmd) {
    ...
    case "newCatchPhrase":
        return new MortyCommand();
    }
```

Commit:
Add Gradle support

Fix gradle error

SeeuSim committed:
Level-1

SeeuSim committed:
Level-2

SeeuSim committed:
Level-3

SeeuSim committed:
A-Classes

SeeuSim committed:
A-Inheritance

SeeuSim committed:
Level-4

SeeuSim committed:
A-TextUiTesting

SeeuSim committed:
A-Exceptions

SeeuSim committed:
Level-5

SeeuSim committed:
A-Enums

SeeuSim committed:
A-Collections

SeeuSim committed:
Level-6

Minor fix for exit code reporting

Fix control flow bug

Fix UI test

Level-6

SeeuSim committed:
Level-7

SeeuSim committed:
Level-8

Search Functions

SeeuSim committed:
Merge branch 'branch-Level-8'

SeeuSim committed:
A-MoreOOP

SeeuSim committed:
A-Packages

SeeuSim committed:
Minor Refactoring
Update names to go with new packaging conventions

SeeuSim committed:
A-JUnit

SeeuSim committed:
A-Jar

SeeuSim committed:
A-CodingStandard

SeeuSim committed:
A-JavaDoc

SeeuSim committed:
Level-9

SeeuSim committed:
Merge branch 'branch-A-CodingStandard'

SeeuSim committed:
Merge branch 'branch-A-JavaDoc'

SeeuSim committed:
Remove unused imports

SeeuSim committed:
Coding Standard fixes

SeeuSim committed:
Level-10
Add basic GUI and logic
- Added JavaFX config to Gradle
- Added JavaFX GUI skeleton and logic
- Refactored package `duke` to `rick`
  - Mainly responsible for app logic
- Refactored testcases
- Added varargs to `Ui::section` to remove redundant `\n` characters

SeeuSim committed:
A-Checkstyle
Add checkstyle config and suppressions
- Add `config/checkstyle` dir containing checkstyle files
- Add config to `build.gradle`
- Fix coding standard issues throughout the entire source code

Signed-off-by: SeeuSim <seeusimong@gmail.com>

SeeuSim committed:
Update README.md

huanghao1998 commented on src/main/java/rick/Rick.java:
```diff
@@ -0,0 +1,52 @@
+    }
+
+    /**
+     * Generate the server UI welcome message.
```
It is better to start the first sentence with Generates, Adds, Returns (Not Return or Returning etc.). <sub>(From Java Coding standard)<sub>

huanghao1998 commented on src/main/java/rick/Parser.java:
```diff
@@ -0,0 +1,128 @@
+                break;
+            }
+        } catch (NumberFormatException e) {
+            return new ErrorCommand(new RickException(
```
It is nice to see you have put line break for long sentences :)

huanghao1998 commented on src/main/java/rick/Ui.java:
```diff
@@ -0,0 +1,96 @@
+ * @author SeeuSim
+ *         AY2223-S2 CS2103T
+ */
+public class Ui {
```
I like the way you implement your Ui methods helps to provide the output, it is very neat

huanghao1998 reviewed (commented):
Your code looks clean and neat, and your OOP is bravo!!!

kennycjy commented on src/main/java/rick/Rick.java:
```diff
@@ -0,0 +1,52 @@
+    }
+
+    /**
+     * Process the command and return the server UI response.
```
Perhaps add a new line after the description, before params to increase readability

kennycjy commented on src/main/java/rick/TaskList.java:
```diff
@@ -0,0 +1,189 @@
+        return ui.section(
+                "Got it. I've added this task:",
+                "  " + input,
+                String.format("Now you have %s task%s in the list.", count, count == 1L ? "" : "s")
```
Good use of new lines to break up the code and increase readability

kennycjy commented on src/main/java/rick/TaskList.java:
```diff
@@ -0,0 +1,189 @@
+     *                                        invalid index was provided.
+     */
+    public String mark(int i) throws TaskListInvalidAccessException {
+        if (i < 0 || i >= this.storage.size()) { //Unassigned, invalid index
```
Nice comment explain if condition

kennycjy commented on src/main/java/rick/Ui.java:
```diff
@@ -0,0 +1,96 @@
+ *         AY2223-S2 CS2103T
+ */
+public class Ui {
+    private static final String mascot =
```
Your constant should be in all caps :D

kennycjy reviewed (commented):
Very creative work, just need a few tweaks to follow conventions :D

SeeuSim committed:
Add Assertions

There are several instances where the Storage may read a task from the
file as `null`. This adds assertions so that those scenarios will not
fail silently.

As the input of data into the file is strictly sanitised, such failures
will only occur if the user's computer errors.

SeeuSim committed:
Fix Code Quality deviations

Found some code quality issues in the codebase and rectified them for
better readability. Refactored all classes and formatted javaDocs.

SeeuSim committed:
Merge pull request #1 from SeeuSim/branch-A-Assertions

Add Assertions

SeeuSim committed:
Merge branch 'master' into branch-A-CodeQuality

SeeuSim committed:
Merge pull request #2 from SeeuSim/branch-A-CodeQuality

Fix Code Quality deviations

SeeuSim committed:
Add CI build file

SeeuSim committed:
Fix workflow

SeeuSim committed:
Fix failing unit tests

SeeuSim committed:
Add MassOps functionality

Users can now manipulate multiple tasks with various flags. These will
be published in detail in the user guide in coming commits, but they are
available for perusal in the Parser class under the manipulateCommand
function.

SeeuSim committed:
Fix MassOps logic

Previously, there were flaws in the storage indexing for the commands
that manipulated (mark, unmark, delete) multiple Tasks at once. This
commit fixes that.

It also compacts all manipulated tasks into one dialog so as not to spam
the chat.

SeeuSim committed:
Fix Code Quality

SeeuSim committed:
Add UI theming

Colors, Background, text bubbles

SeeuSim committed:
Add user guide

SeeuSim committed:
Fix github pages

SeeuSim committed:
Fix github pages theme

SeeuSim committed:
Fix formatting

SeeuSim committed:
Fix theming

SeeuSim committed:
Test new theme

SeeuSim committed:
fix theming

SeeuSim committed:
Configure search

SeeuSim committed:
Set theme to dark mode

SeeuSim committed:
Finalize UI response for delete

SeeuSim committed:
Change header comment for RickTask::containsTerm
